### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by NorbertHeusser**
Thank you for contributing your time to the Mosquitto project!

Before you go any further, please note that we cannot accept contributions if
you haven't signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php).
If you aren't able to do that, or just don't want to, please describe your bug
fix/feature change in an issue. For simple bug fixes it is can be just as easy
for us to be told about the problem and then go fix it directly.

Then please check the following list of things we ask for in your pull request:

- [X] Have you signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php), using the same email address as you used in your commits?
- [X] Do each of your commits have a "Signed-off-by" line, with the correct email address? Use "git commit -s" to generate this line for you.
- [X] If you are contributing a new feature, is your work based off the develop branch?
- [X] If you are contributing a bugfix, is your work based off the fixes branch?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you successfully run `make test` with your changes locally?

-----

### CODE REVIEW

File: apps/db_dump/Makefile [hunk 1]
```diff
@@ -6,24 +6,22 @@ CFLAGS_FINAL=${CFLAGS} ${BROKER_CFLAGS} ${BROKER_CPPFLAGS} -I${R}/include -I${R}
...
+	   stubs.o \
+	   memory_mosq.o \
+	   memory_public.o \
+	   packet_datatypes.o \
+	   persist_read.o \
+	   persist_read_v234.o \
+	   persist_read_v5.o \
+	   property_mosq.o \
+	   topic_tok.o \
+	   utf8_mosq.o
```
Comment (ralight) [inline apps/db_dump/Makefile:18]: I don't want to do this, I want to keep using the files as compiled for the broker.

File: apps/db_dump/Makefile [hunk 2]
```diff
@@ -6,24 +6,22 @@ CFLAGS_FINAL=${CFLAGS} ${BROKER_CFLAGS} ${BROKER_CPPFLAGS} -I${R}/include -I${R}
...
 
 .PHONY: all clean reallyclean
 
 all : mosquitto_db_dump
 
-mosquitto_db_dump : ${OBJS} ${BROKER_OBJS}
+mosquitto_db_dump : ${OBJS}
```
Comment (ralight) [inline apps/db_dump/Makefile:24]: This also should remain as it was, and the related changes further down reverted.

File: apps/db_dump/Makefile [hunk 3]
```diff
@@ -32,16 +30,55 @@ db_dump.o : db_dump.c db_dump.h ${R}/src/persist.h
...
+	${CROSS_COMPILE}${CC} $(CFLAGS_FINAL) -c $< -o $@
+
+utf8_mosq.o : ${R}/lib/utf8_mosq.c
+	${CROSS_COMPILE}${CC} $(CFLAGS_FINAL) -c $< -o $@
+
 reallyclean: clean
 
 clean :
-	-rm -f *.o mosquitto_db_dump *.gcda *.gcno
+	-rm -f  ${OBJS} mosquitto_db_dump *.gcda *.gcno
```
Comment (ralight) [inline apps/db_dump/Makefile:82]: This does need fixing though - it should be `${OBJS} ${BROKER_OBJS}` added, and `*.o` removed.

**Review by ralight — CHANGES_REQUESTED** [review]
Aside from the comments I've made, this looks like a set of sensible changes, thank you. If you remove the parts I've highlighted I'll merge it.

File: apps/db_dump/Makefile (hunk 1 above)
Comment (NorbertHeusser) [reply apps/db_dump/Makefile]: Ok. Simply dropped this commit.

File: apps/db_dump/Makefile (hunk 2 above)
Comment (NorbertHeusser) [reply apps/db_dump/Makefile]: Ok. Simply dropped this commit.

File: apps/db_dump/Makefile (hunk 3 above)
Comment (NorbertHeusser) [reply apps/db_dump/Makefile]: Fixed this one.

**Pull request merged**